data/projects/vulture.md

```diff
@@ -1,57 +1,128 @@
+
+Also, if able to fetch the source range successfully, implement the pathway through which  the metadata flows in and out of API, this would not require much work because  we can easily change item.lineno (int) to item.dead_range (tuple of ints) and can parse them over in the VultureBear.
+
+##### 4.) Detecting unreachable code
+We would first need to identify cases where code cannot be reached. Some of the common ones are:
+
+- `If False`
+- `If True; else`
+- Any code after `return` statement in the block containing return itself.
+- `raise` statement  in `try` block.
+
+Similar constructs would have to be looked onto. The crude form of this would be:
+
+- Analyse the ast’s
+- Look for the if nodes
+- Check the boolean affiliated to it, tracking previous arguments.
+
+
+##### 5.) Implementing a confidence value for  results
+We would need to analyse every construct individually on a case by case basis. For example, we already know that import statements can be predicted with 100% surety (except for * imports, where it would be 0%), but functions often have false-positives.
+
+The confidence value will be alike the ones given below: (The finer grained distinctions will need further discussion)
+
+- `import` -->  `100%`
+- `from foo import *` --> `0%`
+- `variable` --> `<100%`
+- `function` --> `<100%`
+- `class` --> `<100%`
+- `if False` --> `100%`
 
 References:
 
-- [vulture](https://bitbucket.org/jendrikseipp/vulture)
+- [vulture](https://github.com/jendrikseipp/vulture)
+- [Project Proposal](https://docs.google.com/document/d/1gzRH-rdJsiAD-TOEB0O4OqtcnTp0zotcNl6agAuXGHw/edit?usp=sharing)
 
 #### Milestones
 
-##### PREPARATION/BONDING
+##### Pre - Community Bonding Period ( Up Till  May 4)
+- Initial investigation and discussion on what features to be added to API, and start implementing them on the go.
+- Continue discussion with the community on ast and enhanced pyflakes ast - What are the underlying differences, and what can be possible implementations for source range acquisition.
+- Diagnose possible cases which would render the code unreachable.
+- Inculcate a habit of downloading the trending project of the day, running vulture on it, analyzing the results and adding anything suitable to whitelist file. This would help us in maintaining a rich whitelist file.
+- Prioritize all the goals and targets
+
+##### Community Bonding (May 5 - 30)
+- Finalize what should be implemented for source range acquisition on the basis of discussions during pre - community bonding period.
+- Implement API - At this point, we would have a fully functioning API, and we would be ready to incorporate these changes into the VultureBear.
```

Comment:
I guess implementing the API should not be done in the community bonding period. Please revise.